internal/cmd/errors_userfacing.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+// UserFacingError wraps an error to provide better error messages to users.
+// Any non-UserFacingError will be printed as "unhandled errors".
+type UserFacingError struct {
+	// original error
+	Underlying error
+
+	// formatted outputs that users see when running the CLI commands
+	// should be a readable sentence rather than a stacktrace
+	UserFacingMessage string
+
+	// show or hide Underlying error from outputting with the UserFacingMessage
+	ShowUnderlying bool
```

Comment:
+1 though it seems `Underlying` should not be nil since it's needed for `Unwrap`. Instead, if we need to print the raw error, just don't set a message

Reply:
I think I over engineered it for now. It was for anytime when we expect an error (eg. we wouldn't need to print 404 not found, just that 'something wasn't found' on the cli). 
but I agree as well that it's better if we just set underlying as nil and log (debug) it instead, since wherever ShowUnderlying:false was to be used, it's an expected and valid workflow. if user really wants to debug, then they still can.